### PR TITLE
EVG-6546: API server should accept host ID if it matches Jasper credentials ID

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-6546

If the host ID given in the agent request doesn't match a host ID in the database (because it was given the intent host ID when started through user data), try to match it against `Host.JasperCredentialsID`.

### CODE REVIEW

File: model/host/db.go [hunk 1]
```diff
@@ -476,6 +476,17 @@ func ByIds(ids []string) db.Q {
 	})
 }
 
+// FindByJasperCredentialsID finds a host with the given Jasper credentials ID.
+func FindOneByJasperCredentialsID(id string) (*Host, error) {
+	h := &Host{}
+	query := bson.M{JasperCredentialsIDKey: id}
```
Comment (john-m-liu) [inline model/host/db.go]: Don't forget to add an index on this

File: model/host/db.go [hunk 2]
```diff
@@ -476,6 +476,17 @@ func ByIds(ids []string) db.Q {
...
+	err := db.FindOne(Collection, query, db.NoProjection, db.NoSort, h)
+	if adb.ResultsNotFound(err) {
+		return nil, err
+	}
+	return h, err
```
Comment (tychoish) [inline model/host/db.go]: wrap errors.

Comment (tychoish) [reply model/host/db.go]: I feel like these APIs that can return nil, nil are always confusing for users and I don't have a great solution.

Comment (Kimchelly) [reply model/host/db.go]: Wrapped the error and rewrote this slightly for clarity.

This shouldn't return nil, nil - it always returns the error if there is one (unless anser could return nil, nil from `db.FindOne()`).

**Pull request merged**